services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsWaypoint.java

```diff
@@ -34,7 +34,7 @@ public static Builder builder() {
    * @return string with the name of the way the coordinate snapped to
    * @since 1.0.0
    */
-  @NonNull
+  @Nullable
```

Comment:
Yeah, great catch!

Reply:
Ohh wait, the builder is going to be `@NonNull` always, isn't it? 🤔

Reply:
I mean the parameter nullability:
> [at]NonNull
    public abstract Builder name(**[at]Nullable** String name);

Reply:
Wondering if we want to keep it as non-nullable in the `Builder`, I mean, normally if you want to set a `name` is going to be non-null (unless that property could be reset or something around those lines). Is the `Builder` used automatically behind the scenes when parsing? I believe not, because that'd crash when I tested it. Noting that in this particular case, the `property` was missing i.e. it wasn't `null`. If we're going to get `null` by any chance, yeah, the builder should be `@Nullable`.

Reply:
> Is the Builder used automatically behind the scenes when parsing?

It's not, but the builder parameter nullability should always reflect the property nullability.